app/src/main/java/com/arctouch/codechallenge/view/home/viewmodel/HomeActivityViewModel.java

```diff
@@ -0,0 +1,56 @@
+package com.arctouch.codechallenge.view.home.viewmodel;
+
+import androidx.lifecycle.LiveData;
+import androidx.lifecycle.MutableLiveData;
+import androidx.lifecycle.ViewModel;
+import androidx.paging.LivePagedListBuilder;
+import androidx.paging.PageKeyedDataSource;
+import androidx.paging.PagedList;
+
+import com.arctouch.codechallenge.data.MovieRepository;
+import com.arctouch.codechallenge.data.model.Movie;
+
+public class HomeActivityViewModel extends ViewModel {
+
+    private LiveData<PagedList<Movie>> moviePagedList;
+    private LiveData<PageKeyedDataSource<Integer, Movie>> liveDataSource;
+    private MovieRepository movieRepository;
+    private MutableLiveData<MutableLiveData<Throwable>> errorHandlerMutable;
+
+    private static final int PAGE_SIZE = 20;
+
+    public void init(){
+
+        if (moviePagedList != null) {
+            return;
+        }
+
+        movieRepository = MovieRepository.getInstance();
```

Comment:
If you were to unit test this class, how would you mock this repository?

Reply:
It would be a challenge because this repository returns Paging Lib Data sources, that are objects that hold the data that my Movie Lists displays, I tried to search the internet while I was developing this I couldn´t find any valid answer to do this.